### PR TITLE
[docs] Added additional explanations

### PR DESCRIPTION
**Pull request by awelc**
Some of the missing explanations that affect user experience.

### CODE REVIEW

**Review by emmazzz — APPROVED** [review]
Thank you for doing this! There seems to be a little typo of "scenario" that should be fixed.

File: doc/src/build/cli-client.md [hunk 1]
```diff
@@ -998,6 +998,9 @@ Please see the [Move developer documentation](move/index.md) for a
 description on how to [write a simple Move code package](move/write-package.md),
 which we can publish using Sui client's `publish` command.
 
+> **Important:** All calls to functions in the `debug` module must be removed from no-test code
```
Comment (Clay-Mysten) [inline doc/src/build/cli-client.md:1001]: What do we mean by "no-test code" here? I see no other reference on this page.

File: doc/src/build/move/debug-publish.md [hunk 2]
```diff
@@ -21,6 +21,9 @@ debug::print_stack_trace();
 ```
 Alternatively, any call to `abort` or assertion failure will also print the stacktrace at the point of failure.
 
+> **Important:** All calls to functions in the `debug` module must be removed from no-test code
```
Comment (Clay-Mysten) [inline doc/src/build/move/debug-publish.md:24]: Same comment as above. Let's make both references to "no-test code" clear.

**Review by Clay-Mysten — APPROVED** [review]
Thanks, Adam! I made a few small changes in commits here and left a couple of questions in comments.

Please note, we have moved the cli-client.md file to the contribute subdirectory. So I can include your change to that file there.

File: doc/src/build/cli-client.md (hunk 1 above)
Comment (awelc) [reply doc/src/build/cli-client.md:1001]: Added additional explanation in both places.

Comment (Clay-Mysten) [reply doc/src/build/cli-client.md:1001]: > Added additional explanation in both places.

Thanks, Adam! LGTM.

**Comment by Clay-Mysten** [comment]
I'm adding this to the newly moved `cli-client.md` file in:
https://github.com/MystenLabs/sui/pull/4090

**Pull request merged**